### ISSUE TITLE
Crash (assert) related to input from vcl.list

### ISSUE THREAD

**Issue by espebra** [varnish/vagent2#172]
```
Aug 7 10:40:00 hostname varnish-agent[19574]: modules/vcl.c:248: vcl_list_json: Assertion `strlen(pos) > 30' failed. Aborting.
```

I don't currently have information about the versions involved. Seemingly relevant code: https://github.com/varnish/vagent2/blob/master/src/modules/vcl.c#L248

**Comment by Dridi** [comment]
I'm contemplating the idea of using grammars for parsing, because `sscanf` and space-alignment are way to brittle. I haven't found the exact reason for the crash yet, but I suspect it can only happen since Varnish 4.1.2.

This snippet is no longer sufficient on the current master:
https://github.com/varnish/vagent2/blob/f9ad0f0/src/modules/vcl.c#L249-L253

Essentially the `%4s/%4s` part is wrong because the after the slash, it looks like this in Varnish:

```
$ git grep -F /%-
bin/varnishd/cache/cache_vcl.c:         VCLI_Out(cli, "%-10s %5s/%-8s %6u %s",
bin/varnishd/mgt/mgt_vcl.c:                     VCLI_Out(cli, "/%-8s", vp->warm ?
```

FYI the manager code has relatively recently been polished and the single pattern is now split into several (making it slightly harder to understand the pattern with a single glance).

The reason it went from `%4s/%4s` to `%5s/%-8s` is because of the `cooling` temperature. It prints `state/temperature` and both used to be 4 letters each. Now the state is still 4 letters wide (so `%5s` for an extra space before) and the temperature can take up to 7 characters (so `%-8s` for an extra space after).

I don't know yet how I'm going to fix that.

**Comment by Dridi** [comment]
Pinging @carlosabalde in case he wants to subscribe to this issue.

**Comment by carlosabalde** [comment]
Still no idea on how to reproduce this, but since we updated to VCP 4.1.4r1 some servers it already happened a few times.

**Comment by carlosabalde** [comment]
And it happened again in a couple of VCP 4.1.4r1 instances deployed in a different customer. In both cases this happened a few seconds after the worker process was restarted due to a panic.

**Comment by rezan** [comment]
This happened to me yesterday. If anything, change the assert to a check and call failure. Having varnish-agent exit seems to make the problem worse.

**Comment by Dridi** [comment]
@carlosabalde, @rezan, can you please try to run #178, I think @gquintard touched this area in his various cleanups. I don't know when I'll be able to review it, but I vote for a merge if it solves this problem.

**Comment by Dridi** [comment]
@carlosabalde could you please try building #178 and report on overall stability?

**Comment by carlosabalde** [comment]
@Dridi: I'll need to check again with the customers facing this issue if they agree to use a custom build. I'll let you know asap.

**Comment by Dridi** [comment]
@carlosabalde yesterday and today, the changes from @gquintard and more from myself landed in the master branch. Mostly bug fixes and code cleaning.

**Comment by denisbr** [comment]
According to release history, this issue was fixed in the 4.1.2 release of vagent2. 
Based on input from @carlosabalde it sounds like the fix has been verified in production, but I'll leave closing the issue to @Dridi

**Comment by Dridi** [comment]
We can reopen it if it happens again, closing.
